docs/extensions/readme.md

```diff
@@ -34,6 +40,44 @@ The following documents describes AutoRest specific vendor extensions for [OpenA
 - [x-ms-request-id](#x-ms-request-id) - allows to overwrite the request id header name
 - [x-ms-client-request-id](#x-ms-client-request-id) - allows to overwrite the client request id header name
 
+# Metadata Extensions
+
+## x-ms-secret
+
+This extension is used to annotate the secret property. The value type is boolean and the allowed value is either true or false. Secrets should never expose on a GET.
+ 
+**Rule**
+
+* When applying this extensions as: "x-ms-secret": true; means that this property must not exist in response.
```

Comment:
Feels weird we have this just as a "documentation" feature. This should most likely be applied to the generated code but maybe this is something we can just leave to cadl.

Reply:
Our team (Azure Change Analysis) will add new metadata annotations too (x-ms-identifiers).
I think this is not just about documentation, the swagger validation team told me I need to add it here so it will be recognized by the swagger validation tools (at least workflow wise they want me to doc it here).